src/HeapedVectors.jl

```diff
@@ -0,0 +1,156 @@
+__precompile__()
+
+struct HeapedVector{T, F<:Function}  
+    data::Vector{T}
+    by::F
+    function HeapedVector(v::Vector{T}, by::F) where {T, F}
+        new{T, F}(heaping(v, by), by)
+    end
+end
+      
+HeapedVector(data::Vector{T}) where {T} = HeapedVector(data, identity)  
+
+       
+function heaping(v, by) 
+    ar = []
```

Comment:
This array should have an element type which is the same as `v`'s.

Reply:
Or just make a copy of `v` to start with?

Reply:
ok, I am changing it to ar = typeof (v[1])[ ]